pylint/testutils/output_line.py

```diff
@@ -62,39 +76,42 @@ def __init__(self, row, exception):
 class OutputLine(NamedTuple):
     symbol: str
     lineno: int
-    column: str
-    object: Any
+    column: int
+    object: str
     msg: str
-    confidence: interfaces.Confidence
+    confidence: str
 
     @classmethod
-    def from_msg(cls, msg):
+    def from_msg(cls, msg: Message) -> "OutputLine":
         column = cls.get_column(msg.column)
         return cls(
             msg.symbol,
             msg.line,
             column,
             msg.obj or "",
             msg.msg.replace("\r\n", "\n"),
-            msg.confidence.name
-            if msg.confidence != interfaces.UNDEFINED
-            else interfaces.HIGH.name,
+            msg.confidence.name if msg.confidence != UNDEFINED else HIGH.name,
         )
 
     @classmethod
-    def get_column(cls, column):
-        if not PY38_PLUS:
-            return ""  # pragma: no cover
-        return str(column)
+    def get_column(cls, column: str) -> int:
+        if (
+            not PY38_PLUS
+        ):  # This mimicks behaviour of MessagesHandlerMixIn.add_one_message()
+            return 0  # pragma: no cover
+        return int(column)
 
     @classmethod
-    def from_csv(cls, row):
+    def from_csv(cls, row: Union[Tuple[str, ...], List[str], str]) -> "OutputLine":
         try:
-            confidence = row[5] if len(row) == 6 else interfaces.HIGH.name
             column = cls.get_column(row[2])
-            return cls(row[0], int(row[1]), column, row[3], row[4], confidence)
+            if isinstance(row, Iterable) and len(row) == 6:
+                return cls(row[0], int(row[1]), column, row[3], row[4], row[5])
+            if isinstance(row, Iterable) and len(row) == 5:
+                return cls(row[0], int(row[1]), column, row[3], row[4], HIGH.name)
+            raise IndexError
         except Exception as e:
             raise MalformedOutputLineException(row, e) from e
 
-    def to_csv(self):
-        return tuple(self)
+    def to_csv(self) -> Tuple[str, str, str, str, str, str]:
+        return tuple(str(i) for i in self)  # type: ignore # pylint: disable=not-an-iterable
```

Comment:
`pylint` finds that `self` is non-iterable, although I think we can assume it is, right?

Furthermore, `mypy` sees the return value as being `Tuple[str, ...]`. I added checks with a temp value and `isinstance(temp_csv, Tuple) and len(temp_csv) == 6`, but it kept denying the correct size. Eventually I opted to add an ignore.

Reply:
@cdce8p btw, this is a false positive right? Or am I missing some edge case where this is indeed a valid error?